shared/actions/login/saga.js

```diff
@@ -564,6 +564,13 @@ const _deviceTypeMap: {[key: string]: any} = {
   [Constants.codePageDeviceRoleNewPhone]: Types.CommonDeviceType.mobile,
 }
 
+function secretExchangedSaga() {
```

Comment:
why make a function that return the generator vs just having the generator?

Reply:
Just having the generator results in:
```
uncaught at check call: argument [object Generator] is not a function
```
(The style I used matches the other subsaga's definition.)